source/sorted_table_model.ts

```diff
@@ -0,0 +1,198 @@
+import * as Kola from 'kola-signals';
+import { Comparator } from './comparator';
+import { AddRowOperation, MoveRowOperation, Operation, RemoveRowOperation,
+  Transaction, UpdateOperation } from './operations';
+import { SortOrder } from './sort_order';
+import {TableModel} from './table_model';
+import {TransactionLog} from './transaction_log';
+import {TranslatedTableModel} from './translated_table_model';
+
+/** A table that sorts columns. */
+export class SortedTableModel extends TableModel {
+
+  /**
+   * Constructs a model adapting an existing TableModel.
+   * @param model The TableModel to adapt.
+   */
+  constructor(model: TableModel) {
+    super();
+    this.comparator = new Comparator();
+    this.sortOrder = [];
+    for(let i = 0; i < model.columnCount; ++i) {
+      this.sortOrder.push(SortOrder.NONE);
+    }
+    this.sortPriority = [];
+    this.translatedTable = new TranslatedTableModel(model);
+    this.translatedTable.connect(this.handleSourceOperation);
+    this.movesToIgnore = 0;
+    this.transactionLog = new TransactionLog();
+  }
+
+  /**
+   * Marks the beginning of a transaction. In cases where a transaction is
+   * already being processed, then the sub-transaction gets consolidated into
+   * the parent transaction.
+   */
+  public beginTransaction(): void {
+    this.transactionLog.beginTransaction();
+  }
+
+  /** Ends a transaction. */
+  public endTransaction(): void {
+    this.transactionLog.endTransaction();
+  }
+
+  public get rowCount(): number {
+    return this.translatedTable.rowCount;
+  }
+
+  public get columnCount(): number {
+    return this.translatedTable.columnCount;
+  }
+
+  public get(row: number, column: number) {
+    return this.translatedTable.get(row, column);
+  }
+
+  public connect(slot: (operation: Operation) => void):
+      Kola.Listener<Operation> {
+    return this.transactionLog.connect(slot);
+  }
+
+  /** 
+   * Sorts the table using @param column as the column with the highest sort
+   * priority. All previous sortOrders remain and are pushed down a rank.
+   * @param column - The column that will now have the highest priority
+   * @param sortOrder - The sort order of column.
+   */
+  public updateSort(column: number, sortOrder: SortOrder) {
```

Comment:
Avoid inconsistencies in naming, it results in ambiguities:

```updateSort``` -> ```updateSortOrder```

Reply:
If you prefer a short form, then eliminate adjectives and keep the noun instead of eliminating the noun and keeping the adjective:

```updateOrder``` is preferable to ```updateSort```.